tasks/bin/cross-ruby.rake

```diff
@@ -126,6 +126,8 @@ file makefile_in => [makefile_in_bak] do |t|
   }
 end
 
+task "prepare-sources" => makefile_in
```

Comment:
This approach isn't enough.
Parallel `rake-compiler prepare-sources` may create a broken prepared source:

```bash
rm -rf ~/.rake-compiler
for host in i686-w64-mingw32 x86_64-w64-mingw32; do
  ruby -Ilib bin/rake-compiler cross-ruby VERSION=2.6.0 HOST=$host &
done
wait
ls -lah ~/.rake-compiler/sources
# total 28M
# drwxr-xr-x  3 kou kou 4.0K Dec 25 06:27 .
# drwxr-xr-x  3 kou kou 4.0K Dec 25 06:27 ..
# drwxr-xr-x 27 kou kou 4.0K Dec 25 06:27 ruby-2.6.0
# -rw-r--r--  1 kou kou  14M Dec 25  2018 ruby-2.6.0.tar.bz2
# -rw-r--r--  1 kou kou  14M Dec 25  2018 ruby-2.6.0.tar.bz2.1
```

Reply:
Sure, in `rake-compiler-dock` I execute the task `prepare-sources`sequentially for all rubies. Afterwards `cross-ruby` is executed in parallel, which works since all source related tasks are already done and skipped.

I didn't try to run the download and extract part in parallel. Maybe locking the source directory or a file inside would work. Is this something I should investigate and you would consider merging?

Reply:
Ah, sorry. I misunderstood.

I'm positive for parallel build support. But I think that creating a lock file instead of using `flock` approach is better. (I didn't try this yet. So we may not be able to use lock file approach.)

Reply:
Or is it better that we add support for `rake-compiler cross-ruby VERSIONS=i686-w64-mingw32-2.6.0:x86_64-w64-mingw32-2.7.0` syntax and use Rake's parallel build feature?

Reply:
`rake-compiler cross-ruby VERSIONS=i686-w64-mingw32-2.6.0:x86_64-w64-mingw32-2.7.0` implemented per multitask sounds good.

Is there anything wrong in using `File#flock`? Why should we create a separate lock file instead? Locking the vulnerable file by the operation system is the cleanest approach IMHO.